File: LeetCode/172. Factorial Trailing Zeroes/Solution.java
public class Solution {
    public int trailingZeroes(int n) {
        int sum = 0;//the ans is n/5 + n/25 + n/125 + ...
        int t = 5;
        while (t <= n) {
            sum += n / t;
            n /= 5;
        }
        return sum;
    }
}

File: LeetCode/172. Factorial Trailing Zeroes/solution.py
from math import log


class Solution(object):
    def trailingZeroes(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans, pace = 0, 5
        while n >= pace:
            ans += n // pace
            pace *= 5
        return ans

    def trailingZeroes1(self, n):
        ans = 0
        while n:
            ans += n / 5
            n /= 5
        return ans
